src/screens/MapScreen/MapScreen.tsx

```diff
@@ -0,0 +1,54 @@
+import MapView from "react-native-maps";
+import MapViewDirections from "react-native-maps-directions";
+
+import { GlobalStyles, locations } from "../../constants";
+import { View } from "react-native";
+import { DefaultButton } from "../../components";
+import { useData } from "./useData";
+export const MapScreen = () => {
+  const {
+    handlePressRecommended,
+    recommendedLocations,
+    mapRef,
+    userCoordinate,
+    destination,
+    renderMarkers,
+  } = useData();
+
+  return (
+    <View style={{ flex: 1 }}>
+      <MapView
+        ref={mapRef}
+        zoomEnabled
+        followsUserLocation
+        showsMyLocationButton
+        showsUserLocation
+        style={{ height: "100%", width: "100%" }}
+        provider="google"
+        initialRegion={{
+          ...locations[0].coordiante,
+          latitudeDelta: 0.2,
+          longitudeDelta: 0.2,
+        }}
+      >
+        <MapViewDirections
+          origin={userCoordinate}
+          destination={destination}
```

Comment:
Missing conditional for no destination,
throws probably the following errors:
 WARN  MapViewDirections Error: Error on GMAPS route request: NOT_FOUND
 WARN  MapViewDirections Error: Error on GMAPS route request: ZERO_RESULTS